lib/finch/http1/pool.ex

```diff
@@ -118,35 +137,57 @@ defmodule Finch.HTTP1.Pool do
     :ok
   end
 
+  @impl Finch.Pool
+  def get_pool_status(finch_name, shp) do
+    case Finch.PoolManager.get_pool_count(finch_name, shp) do
+      nil ->
+        {:error, :not_found}
+
+      count ->
+        result = Enum.map(1..count, &PoolMetrics.get_pool_status(finch_name, shp, &1))
+
+        if Enum.all?(result, &match?({:ok, _}, &1)),
```

Comment:
Wouldn't partial results still be useful? For example, if a connection process were to restart, would that cause this function to return `{:error, :not_found}`? If so that error message could be misleading as pools do exist, but one or more happened to be unavailable at that moment. Plus, I think I would still like to get metrics for the rest of the pools in that case rather than none. Interested to hear your reasoning though.

Reply:
I think you are right! My initial thought was that we probably never get into this state where some counters are not available. In fact with the current implementation the only way to this happen is on pool startup.

AFAICT is impossible to this situation happen otherwise because we are not cleaning (erasing de persistent_term) the state when the pool is terminated for any reason. So, even if we call `get_pool_status/2` between restarts the worst that could happen is serving stale metrics from the restarted pool until the new pool updates de persistent term on init.

Since I think would be desirable to leverage the terminate_pool callback that I've implemented on nimble_pool https://github.com/dashbitco/nimble_pool/pull/38 in order to do some clean up on pool termination, I think would be better to already go as you say and serve all the metrics we have even if some metrics are not available between restarts. Gonna fix this soon! Thanks for the feedback.